cmd/kubectl-gadget/deploy.go

```diff
@@ -100,6 +100,10 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: spec.nodeName
+          - name: GADGET_CONTAINER_UID
```

Comment:
It should be GADGET_POD_UID since this is the uid of the pod.

Reply:
Fixed.